### PR TITLE
feat: update v11 step 3 to React 18

### PR DESCRIPTION
**Pull request by alisonjoseph**
Ref #https://github.com/carbon-design-system/carbon/issues/13630

#### Changelog

**New**

- Add react testing library

**Changed**

- Upgrade to @carbon/react@1.32.0
- Upgrade to React 18
- Simplified tests
- Update index.js to use `createRoot`

**Removed**

- remove enzyme
- remove enzyme-adapter-react-16

**Note: Tests should fail, they will only pass after the user finishes step 3 and adds a table with pagination**

### CODE REVIEW

File: src/App.test.js [hunk 1]
```diff
@@ -1,31 +1,15 @@
...
+it('renders a table with pagination', async () => {
+  const { container } = render(<RepoPage />);
+  expect(container.getElementsByClassName('.cds--pagination').length).toBe(0);
+  render(<RepoPage />);
+  await waitFor(
+    () => expect(screen.getByText('Items per page')).toBeInTheDocument(),
+    {
+      timeout: 1000,
+    }
+  );
```
Comment (francinelucca) [inline src/App.test.js:14]: I would expect this to pass when testing it from the step 4 branch but here's the error I get when doing that:
image

**Review by francinelucca — CHANGES_REQUESTED** [review]

File: src/App.test.js [hunk 2]
```diff
@@ -1,31 +1,15 @@
...
+import { render, screen, waitFor } from '@testing-library/react';
+
+it('renders a table with pagination', async () => {
+  const { container } = render(<RepoPage />);
+  expect(container.getElementsByClassName('.cds--pagination').length).toBe(0);
+  render(<RepoPage />);
+  await waitFor(
+    () => expect(screen.getByText('Items per page')).toBeInTheDocument(),
+    {
+      timeout: 1000,
```
Comment (alisonjoseph) [inline src/App.test.js:12]:
```suggestion
      timeout: 2000,
```

File: src/App.test.js (hunk 1 above)
Comment (alisonjoseph) [reply src/App.test.js]: @francinelucca just increased the timeout to 2000 and they passed for me, let me know if you see something different. 🙏

Comment (francinelucca) [reply src/App.test.js]: works now!

**Review by francinelucca — APPROVED** [review]
🚀

**Review by tay1orjones — APPROVED** [review]

**Review by andreancardona — APPROVED** [review]

**Pull request merged**